contribute/backend/upgrading-go-version.md

```diff
@@ -0,0 +1,17 @@
+# Upgrading Go Version
+
+Notes on upgrading Go version.
+
+Example PR: https://github.com/grafana/grafana/pull/79329
+
+## The main areas that need to change during the upgrade are:
+- https://github.com/grafana/grafana/blob/d8ecea4ed93efb2e4d64a5ee24bc08f3805f413d/scripts/drone/variables.star#L6
```

Comment:
I think I can include here the need to re-generate the `drone.yml` by running `make drone`

Reply:
Good point, added https://github.com/grafana/grafana/pull/79329/commits/0769094faf9464da0d49c9d0842f3fde6fd5f953